CMakeLists.txt

```diff
@@ -277,7 +275,14 @@ ENDIF()
 # Always enable debug sync for debug builds.
 SET(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DENABLED_DEBUG_SYNC")
 SET(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -DENABLED_DEBUG_SYNC")
- 
+
+IF(MSVC)
+  SET(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -D_ITERATOR_DEBUG_LEVEL=1")
+ELSE()
+  # Enable extra checks in GNU libstdc++
+  SET(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -D_GLIBCXX_DEBUG -D_GLIBCXX_ASSERTIONS")
```

Comment:
It looks like we must not enable this for too old GCC:
```
/usr/include/c++/8/debug/helper_functions.h:92:17: note:   ‘const List<Key_part_spec>::Iterator’ is not derived from ‘const __gnu_debug::_Safe_iterator<_IteratorL, _Sequence>’
       if (__lhs == __rhs)
           ~~~~~~^~~~~~~~
```
There is no issue to compile the embedded server with GCC 13.2.0, and `./mtr --embedded` would find just the expected failures.